SPEC.md

```diff
@@ -455,36 +696,74 @@ lines](https://www.unicode.org/versions/Unicode13.0.0/ch05.pdf):
 
 Note that for the purpose of new lines, CRLF is considered _a single newline_.
 
+### Disallowed Literal Code Points
+
+The following code points may not appear literally anywhere in the document.
+They may be represented in Strings (but not Raw Strings) using `\u{}`.
+
+* The codepoints `U+0000-0008` or the codepoints `U+000E-001F`  (various
+  control characters).
+* `U+007F` (the Delete control character).
+* Any codepoint that is not a [Unicode Scalar
+  Value](https://unicode.org/glossary/#unicode_scalar_value) (`U+D800-DFFF`).
+* `U+200E-200F`, `U+202A-202E`, and `U+2066-2069`, the [unicode
+  "direction control"
+  characters](https://www.w3.org/International/questions/qa-bidi-unicode-controls)
+
 ## Full Grammar
 
+This is the full official grammar for KDL and should be considered
+authoritative if something seems to disagree with the text above. The [grammar
+language syntax](#grammar-language) is defined below.
+
 ```
-nodes := linespace* (node nodes?)? linespace*
+document := bom? nodes
+
+nodes := ('/-' plain-node-space* node)? (line-space* node)* line-space*
```

Comment:
If I understand this correctly, the following is valid KDL

```kdl
one
two
three
```

but this isn't:

```kdl
one
/-two
three
```

because `line-space` requires at least one `plain-line-space` before any slash-dashed node and the newline after node `one` is already consumed as the `node-terminator`.

Am I missing something or is there something wrong here?

Reply:
Is that right? I'm not sure how to resolve it off the top of my head. I could use some help 😅

Reply:
Doesn't the `*` operator resolve this? You need 0-or-more additional linespace. The bigger problem I'm seeing is that there's no repetition that I can see for slashdash nodes.

Reply:
@Lucretiel `line-space` contains the slashdash:

```
line-space := plain-line-space+ ('/-' plain-node-space* node)?
```

I think the slashdash in `nodes` was added to allow documents to start with a slashdash? 

@zkat  Could we define `line-space` as follows?

```
line-space := plain-line-space+ | '/-' plain-node-space* node
```

`node` already forces a `node-terminator` at the end, so I don't think a `plain-line-space` is necessary before a slash-dashed node: `one;/-two;three` should be valid, just like `one;two;three` already is.

(This change would also allow removing the `('/-' plain-node-space* node)?` at the start of `nodes`)